Source/Core/VideoBackends/Vulkan/CMakeLists.txt

```diff
@@ -28,14 +28,28 @@ set(LIBS
   common
 )
 
+add_dolphin_library(videovulkan "${SRCS}" "${LIBS}")
+
 # Only include the Vulkan headers when building the Vulkan backend
-include_directories(${CMAKE_SOURCE_DIR}/Externals/Vulkan/Include)
+target_include_directories(videovulkan PRIVATE ${CMAKE_SOURCE_DIR}/Externals/Vulkan/Include)
 
 # Silence warnings on glslang by flagging it as a system include
-include_directories(SYSTEM ${CMAKE_SOURCE_DIR}/Externals/glslang/glslang/Public)
-include_directories(SYSTEM ${CMAKE_SOURCE_DIR}/Externals/glslang/SPIRV)
+target_include_directories(videovulkan SYSTEM PRIVATE ${CMAKE_SOURCE_DIR}/Externals/glslang/glslang/Public)
+target_include_directories(videovulkan SYSTEM PRIVATE ${CMAKE_SOURCE_DIR}/Externals/glslang/SPIRV)
 
 # Link against glslang, the other necessary libraries are referenced by the executable.
-add_dolphin_library(videovulkan "${SRCS}" "${LIBS}")
 target_link_libraries(videovulkan PRIVATE glslang)
 
+# Determine which surface backend to use.
+if(WIN32)
+  target_compile_definitions(videovulkan PRIVATE -DVK_USE_PLATFORM_WIN32_KHR)
+elseif(ANDROID)
+  target_compile_definitions(videovulkan PRIVATE -DVK_USE_PLATFORM_ANDROID_KHR)
+elseif(USE_X11)
+  # When building for X11, we need to link to libX11-xcb.
+  find_package(X11_XCB REQUIRED)
+  target_compile_definitions(videovulkan PRIVATE -DVK_USE_PLATFORM_XCB_KHR ${X11_XCB_DEFINITIONS})
```

Comment:
You can't have -D in front of defines with target_compile_definitions().
It's only required for add_definitions() since there wasn't any other way to pass flags in old CMake.

I kind of liked it better in the header file with the preprocessor as it reduces the complexity in the build system for all platforms.